pkg/operator/ceph/object/rgw.go

```diff
@@ -130,14 +132,16 @@ func (c *clusterConfig) startRGWPods(realmName, zoneGroupName, zoneName string,
 		resourceName := fmt.Sprintf("%s-%s-%s", AppName, c.store.Name, daemonLetterID)
 
 		rgwConfig := &rgwConfig{
-			ResourceName:   resourceName,
-			DaemonID:       daemonName,
-			Realm:          realmName,
-			ZoneGroup:      zoneGroupName,
-			Zone:           zoneName,
-			Auth:           c.store.Spec.Auth,
-			Protocols:      c.store.Spec.Protocols,
-			KeystoneSecret: keystoneSecret,
+			ResourceName:      resourceName,
+			DaemonID:          daemonName,
+			Realm:             realmName,
+			ZoneGroup:         zoneGroupName,
+			Zone:              zoneName,
+			Auth:              c.store.Spec.Auth,
+			Protocols:         c.store.Spec.Protocols,
+			KeystoneSecret:    keystoneSecret,
+			FailureDomain:     c.getRGWFailureDomain(),
+			ReadReplicaPolicy: c.store.Spec.Gateway.RgwConfig[radosReadReplicaPolicy],
```

Comment:
The intent of the `rgwConfig` (and similar) options for config overrides is that they are simple passthroughs. I don't think we should be using these to inform more advanced behaviors. I think it would be best to have a dedicated opt-in for controlling the init container behavior.

Reply:
reverted back to using a new API to set the readAffinity: 
```
type RgwReadAffinity struct {
	Type string `json:"type"`
	FailureDomain string `json:"failureDomain,omitempty"`
}
```